### ISSUE TITLE
`Cannot move` false positive when using default generic type arguments

### ISSUE THREAD

**Issue by aldanor** [intellij-rust/intellij-rust#9923]
Here's an example of "cannot move" false positive which, for whatever reason, seems to be caused by the default generic argument IIUC (if `Z = Y<T>` is removed, it seems to disappear):

```rust
struct Y<T>(usize, std::marker::PhantomData<T>);

impl<T> Clone for Y<T> {
    fn clone(&self) -> Self {
        Self(self.0, std::marker::PhantomData)
    }
}

impl<T> Copy for Y<T> {}

struct X<T, Z = Y<T>> { // <-- ?
    x: Z,
    m: std::marker::PhantomData<T>,
}

fn f<T>(x: &X<T>) -> Y<T> {
    x.x // <-- "cannot move", false positive
}
```

## Environment

* **IntelliJ Rust plugin version:** 0.4.185.5086-223
* **Rust toolchain version:** 1.66
* **IDE name and version:** CLion 2022.3
* **Operating system:** macOS

**Comment by aldanor** [comment]
@Undin @vlad20012 here's another one (could be the same thing?)

```rust
trait IntoEnumIterator: Sized {
    type Iterator: Iterator<Item = Self>;
    fn iter() -> Self::Iterator;
}

fn f<T: IntoEnumIterator + Copy>() {
    for value in T::iter() {
        g(value);
        g(value); // <-- false positive: "Use of moved value"
    }
}

fn g<T>(_: T) {}
```
